internal/platform/utimes_darwin.go

```diff
@@ -0,0 +1,33 @@
+package platform
+
+import (
+	"syscall"
+	"unsafe"
+)
+
+func futimens(fd uintptr, atimeNsec, mtimeNsec int64) error {
+	tv := []syscall.Timespec{
+		syscall.NsecToTimespec(atimeNsec),
+		syscall.NsecToTimespec(mtimeNsec),
+	}
+
+	_, _, e1 := syscall_syscall6(libc_futimens_trampoline_addr, fd, uintptr(unsafe.Pointer(&tv[0])), 0, 0, 0, 0)
+	if e1 != 0 {
+		return e1
+	}
+	return nil
+}
+
+// libc_futimens_trampoline_addr is the address of the libc_futimens_trampoline symbol, defined in utimes_darwin.s
+// we use this value to invoke the syscall through syscall_syscall6 imported below
+var libc_futimens_trampoline_addr uintptr
+
+// Imports the futimens symbol from libc as libc_futimens
+//go:cgo_import_dynamic libc_futimens futimens "/usr/lib/libSystem.B.dylib"
```

Comment:
so go:cgo_import_dynamic doesn't need cgo?

Reply:
will do a follow-up doc change, as @mathetake said "regardless of CGO_ENABLED, Go always uses “CGO” mechanism in the runtime layer (of darwin), because that's where Apple provides the compatibility layer." put another way, you can't build a statically linked binary for darwin.